hazelcast/src/main/java/com/hazelcast/map/impl/MapContainer.java

```diff
@@ -129,6 +129,9 @@ public RecordFactory createNew(Void notUsedArg) {
                     case OBJECT:
                         recordFactory = new ObjectRecordFactory(mapConfig, serializationService);
                         break;
+                    case NATIVE:
+                        throw new IllegalArgumentException("Native storage format is supported in Hazelcast Enterprise only. "
```

Comment:
Similar issue here. So the improvement is good, but I would make the error messages more consistent.